File: practica2/c++/main.cc
#include <iostream>
#include "suma.h"
#include "read.h"
#include "write.h"
#include "push.h"
#include "dup.h"
#include "jumpif.h"
#include "mul.h"
#include "swp.h"
#include "over.h"
#include "progSuma.h"
#include "progCuenta.h"
#include "progFactorial.h"

int main(int argc, char* argv[])
{
	programa* pruebas[3];
	cout << "Programa de suma: " << endl;
	pruebas[0] = new progSuma();
	cout << "Programa: " << endl;
	pruebas[0]->listar();
	cout << endl;
	cout << "Ejecución: " << endl;
	pruebas[0]->ejecutar();
	cout << endl << "Programa de cuenta atrás: " << endl;
	pruebas[1] = new progCuenta();
	cout << "Programa: " << endl;
	pruebas[1] -> listar();
	cout << endl;
	cout << "Ejecución: " << endl;
	pruebas[1] -> ejecutar();
	cout << endl;
	cout << "Programa factorial: " << endl;
	pruebas[2] = new progFactorial();
	cout << "Programa: " << endl;
	pruebas[2] -> listar();
	cout << endl;
	cout << "Ejecución: " << endl;
	pruebas[2] -> ejecutar();
	cout << endl;
}

File: practica2/java/Main.java


public class Main
	{
		public static void main(String[] args)
		{
			Programa[] pruebas = new Programa[3];
			System.out.println("Programa de suma:");
			pruebas[0] = new ProgSuma();
			System.out.println("Programa:");
			pruebas[0].listar();
			System.out.println("");
			System.out.println("Ejecución:");
			pruebas[0].ejecutar();
			System.out.println("");
			System.out.println("Programa de cuenta:");
			pruebas[1] = new ProgCuenta();
			System.out.println("Programa:");
			pruebas[1].listar();
			System.out.println("");
			System.out.println("Ejecucion:");
			pruebas[1].ejecutar();
			System.out.println("");
			System.out.println("Programa de factorial:");
			pruebas[2] = new ProgFactorial();
			System.out.println("Programa:");
			pruebas[2].listar();
			System.out.println("");
			System.out.println("Ejecucion:");
			pruebas[2].ejecutar();
			System.out.println("");
		}
	}
